### PR TITLE
storage: sort InternalReplicas before comparing

### PR DESCRIPTION
**Pull request by jeffrey-xiao**
Fixes an issue where the in-memory copy of RangeDescriptor is out-of-sync with the copy in kv.

Should resolve the failure in this https://github.com/cockroachdb/cockroach/issues/36983#issuecomment-507027174

The import step of `tpcc/mixed-headroom/n5cpu16` passes on `2811e0f` + #38465 + this PR.

cc @asubiotto

### CODE REVIEW

File: pkg/storage/replica_command.go [hunk 1]
```diff
@@ -1255,6 +1255,17 @@ func conditionalGetDescValueFromDB(
 			return nil, errors.Wrap(err, "decoding current range descriptor value")
 		}
 	}
+	// TODO(jeffreyxiao): This is hacky fix to ensure that we don't fail the
```
Comment (ajwerner) [inline pkg/storage/replica_command.go]: Either `This is a hacky fix is to ensure` or `This hacky fix is to ensure`?

File: pkg/storage/replica_command.go [hunk 2]
```diff
@@ -1255,6 +1255,17 @@ func conditionalGetDescValueFromDB(
...
+	// conditional get because of the ordering of InternalReplicas. Calling
+	// Replicas() will sort the list of InternalReplicas as a side-effect. The
```
Comment (ajwerner) [inline pkg/storage/replica_command.go]: Can you add some more details about when and why this sorting became necessary?

**Review by jeffrey-xiao — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/38593)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ajwerner and @nvanbenschoten)

___
*[pkg/storage/replica_command.go, line 1258 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/38593#-LijJ4Nq7-PN6bQpXvt2-r1-1258:-LijL_sqBXiJTUFZlSIu:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/a7c860d24eafa7f980370db1d0c6ce4192411c25/pkg/storage/replica_command.go#L1258)):*
<details><summary><i>Previously, ajwerner wrote…</i></summary><blockquote>

Either `This is a hacky fix is to ensure` or `This hacky fix is to ensure`?
</blockquote></details>

Done.
___
*[pkg/storage/replica_command.go, line 1260 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/38593#-LijJ4Nq7-PN6bQpXvt2-r1-1260:-LijLaAqElW10c9YLYOG:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/a7c860d24eafa7f980370db1d0c6ce4192411c25/pkg/storage/replica_command.go#L1260)):*
<details><summary><i>Previously, ajwerner wrote…</i></summary><blockquote>

Can you add some more details about when and why this sorting became necessary?
</blockquote></details>

Done.

**Review by nvb — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/38593#-:-LijMHWKAfkNdvO0Yoc2:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @ajwerner, @jeffrey-xiao, and @nvanbenschoten)

___
*[pkg/storage/replica_command.go, line 1269 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/38593#-LijMOMV9HTYr56_tzzW:-LijMOMW7EjnigHUBaeb:b-9kwqpl) ([raw file](https://github.com/cockroachdb/cockroach/blob/2ed5d152ef048c4c8094c9e928c72a6f1fe2b38b/pkg/storage/replica_command.go#L1269)):*
> ```Go
> 	// ReplicaDescriptors.Learners().
> 	existingDesc.Replicas()
> 	expectation.Replicas()
> ```

```
existingDesc.Replicas() // for sorting side-effect
expectation.Replicas() // for sorting side-effect
```

**Review by ajwerner — APPROVED** [review]
LGTM2

**Comment by jeffrey-xiao** [comment]
TFTRs!

bors r+

**Pull request merged**